app/models/manageiq/providers/vmware/inventory/parser/network_manager.rb

```diff
@@ -0,0 +1,143 @@
+class ManageIQ::Providers::Vmware::Inventory::Parser::NetworkManager < ManageIQ::Providers::Vmware::Inventory::Parser
+  def parse
+    cloud_networks
+    cloud_subnets
+    network_routers
+    network_ports
+    floating_ips
+  end
+
+  def cloud_networks
+    collector.vdc_networks.each do |vdc_network|
+      parse_network(vdc_network, cloud_network_vdc_type)
+    end
+    collector.vapp_networks.each do |vapp_network|
+      parse_network(vapp_network, cloud_network_vapp_type)
+    end
+  end
+
+  def cloud_subnets
+    collector.vdc_networks.each do |vdc_subnet|
+      parse_network_subnet(vdc_subnet)
+    end
+    collector.vapp_networks.each do |vapp_subnet|
+      parse_network_subnet(vapp_subnet)
+    end
+  end
+
+  def network_routers
+    collector.network_routers.each do |router|
+      network_router = persister.network_routers.find_or_build("#{router[:network_id]}---#{router[:parent_net].id}").assign_attributes(
+        :name          => "Router #{router[:parent_net].name} -> #{router.dig(:net_conf, :networkName)}",
+        :cloud_network => persister.cloud_networks.lazy_find(router[:parent_net].id),
+      )
+
+      cloud_subnet = persister.cloud_subnets.find("subnet-#{router[:network_id]}")
+      cloud_subnet.network_router = network_router
+    end
+  end
+
+  def network_ports
+    collector.nics.each do |port|
+      network_port = persister.network_ports.find_or_build(port_id(port)).assign_attributes(
+        :name        => port_name(port),
+        :device_ref  => port[:vm].id,
+        :device      => port[:vm],
+        :mac_address => port.dig(:MACAddress)
+      )
+
+      network_id = read_network_name_mapping(port[:vm].orchestration_stack.ems_ref, port.dig(:network))
+      network = persister.cloud_networks.find(network_id)
+
+      next if network.nil?
```

Comment:
what does this say, if there is no network, we can't find the subnet? It should still be ok to create the ```cloud_subnet_network_ports```, if there is ```port[:IpAddress]``` we want to show